src/composables/useProductQueue.ts

```diff
@@ -0,0 +1,169 @@
+import { ref, computed } from 'vue';
+import { useStore } from 'vuex';
+import { TransferOrderService } from '@/services/TransferOrderService';
+import { ProductService } from '@/services/ProductService';
+import { StockService } from '@/services/StockService';
+import { hasError, showToast } from '@/utils';
+import { translate } from "@hotwax/dxp-components";
+import logger from '@/logger';
+
+/**
+ * Sequential product addition queue to prevent API deadlocks.
+ * 
+ * Process:
+ * 1. Products are added to queue via addProductToQueue()
+ * 2. Queue processes items sequentially (one API call at a time)
+ * 3. pendingProductIds tracks products being added for UI feedback
+ * 4. Prevents duplicate additions and API conflicts
+ * 
+ * States:
+ * - addQueue: Items waiting to be processed
+ * - pendingProductIds: Products currently being added (for UI)
+ * - isProcessing: Whether queue is actively processing items
+ */
+
+export function useProductQueue() {
+  const store = useStore();
+  
+  const addQueue = ref([]) as any;
+  const isProcessing = ref(false);
+  const pendingProductIds = ref(new Set());
+  
+  const currentOrder = computed(() => store.getters['transferorder/getCurrent']);
+
+  // Helper function to check if product is in order
+  const isProductInOrder = (productId: string ) => {
+    return currentOrder.value?.items?.some((item: any) => item.productId === productId);
+  };
+
+  // Helper function to check if product is being processed
+  const isProductBeingProcessed = (productId: string ) => {
+    return pendingProductIds.value.has(productId) || isProductInOrder(productId);
+  };
+
+  const addProductToQueue = (itemToAdd: any) => {
+    const { product } = itemToAdd;
+    
+    if (!product?.productId || !itemToAdd.orderId) {
+      logger.error('Missing product data or orderId');
+      return;
+    }
+    
+    // Skip if already in order or being processed
+    if (isProductBeingProcessed(product.productId)) {
+      return;
+    }
+    
+    pendingProductIds.value.add(product.productId);
+    
+    addQueue.value.push({
+      ...itemToAdd,
+      scannedId: itemToAdd.scannedId
+    });
```

Comment:
Can't we do it like:


```suggestion
    addQueue.value.push(itemToAdd);
```